mesheryctl/internal/cli/root/relationships/list.go

```diff
@@ -0,0 +1,198 @@
+// limitations under the License.
+
+package relationships
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/eiannone/keyboard"
+	"github.com/fatih/color"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// represents the mesheryctl exp relationships list command
+var listRelationshipsCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List registered relationships",
+	Long:  "List all relationships registered in Meshery Server",
+	Example: `
+	View list of relationship
+    mesheryctl exp relationship list
+    View list of relationship with specified page number (25 relationships per page)
+    mesheryctl exp relationship list --page 2
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		// Check prerequisites for the command here
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	Args: func(_ *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp relationship list \nRun 'mesheryctl exp relationship list --help' to see detailed help message"
+		if len(args) != 0 {
+			return utils.ErrInvalidArgument(errors.New(errMsg))
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 0 {
+			return errors.New(utils.RelationshipsError("this command takes no arguments\n", "list"))
+		}
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		var url string
+		if cmd.Flags().Changed("page") {
+			url = fmt.Sprintf("%s/api/meshmodels/relationships?page=%d", baseUrl, pageNumberFlag)
+		} else {
+			url = fmt.Sprintf("%s/api/meshmodels/relationships?pagesize=all", baseUrl)
+		}
+		req, err := utils.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		resp, err := utils.MakeRequest(req)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		// defers the closing of the response body after its use, ensuring that the resources are properly released.
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		relationshipsResponse := &models.MeshmodelRelationshipsAPIResponse{}
+		err = json.Unmarshal(data, relationshipsResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		header := []string{"Entity Details", "Type"}
```

Comment:
Please refer [here](https://github.com/meshery/meshery/issues/9844#issuecomment-1883215882) to see the columns we need to display.

Reply:
> Please refer [here](https://github.com/meshery/meshery/issues/9844#issuecomment-1883215882) to see the columns we need to display.

Alright, thanks